Lesson29/Gokhan/script.js

```diff
@@ -0,0 +1,163 @@
+// 1. Array Destructuring
+// Given the following array, use array destructuring to assign the values of x, y, and z to their respective variables:
+const coordinates = [10, 20, 30];
+const [x, y, z] = coordinates;
+console.log("Ex1: ", x, y, z);
+
+
+
+// 2. Object Destructuring
+// Given the following object, use object destructuring to assign the values of name and age to their respective variables:
+const personE14 = {
+	name: 'John Doe',
+	age: 25,
+};
+
+const { name: fullName, age } = personE14;
+
+console.log("Ex2: ", fullName, age);
+
+
+
+// 3. Array Destructuring with Default Values
+// Given the following array, use array destructuring with default values to assign the values of a, b, and c, with default values of 1, 2, and 3 respectively:
+let numbers = [4, undefined, true];
+
+const [a = 1, b = 2, c = 3] = numbers;
+
+console.log("Ex3: ", a, b, c); // 4, 2, true
+
+
+
+// 4. Object Destructuring with Renaming
+// Given the following object, use object destructuring with renaming to assign the value of name to a variable named fullName:
+const personE16 = {
+	name: 'Jane Doe',
+};
+
+const { name: fullName1} = personE16;
+console.log("Ex4: ", fullName1);
+
+
+
+// 5. Nested Object Destructuring
+// Given the following nested object, use object destructuring to assign the values of name, age, and city to their respective variables:
+const personE17 = {
+	name: 'John Doe',
+	age: 30,
+	address: {
+		city: 'New York',
+	},
+};
+
+const { name, age: age1, address } = personE17;
+const { city } = address;
```

Comment:
This can also be done in one line: `const { name, age: age1, address: { city } } = personE17;`. No need to do changes, note for the future.